sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CertificateValidationCheck.cs

```diff
@@ -0,0 +1,299 @@
+                    if (methodParamLookup.TryGetSymbolParameter(param, out var argument))
+                    { //For Lambda expression extract location of the parentheses only to separate them from secondary location of "true"
+                        var primaryLocation = ((argument.Expression is ParenthesizedLambdaExpressionSyntax Lambda) ? (SyntaxNode)Lambda.ParameterList : argument).GetLocation();
+                        TryReportLocations(new InspectionContext(c), primaryLocation, argument.Expression);
+                    }
+                }
+            }
+        }
+
+        private void TryReportLocations(InspectionContext c, Location primaryLocation, ExpressionSyntax expression)
+        {
+            var locations = ArgumentLocations(c, expression);
+            if (!locations.IsEmpty)
+            {   //Report both, assignemnt as well as all implementation occurances
+                c.Context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, primaryLocation, additionalLocations: locations));
+            }
+        }
+
+        private static bool IsValidationDelegateType(ITypeSymbol type)
+        {
+            if (type.Is(KnownType.System_Net_Security_RemoteCertificateValidationCallback))
+            {
+                return true;
+            }
+            else if (type is INamedTypeSymbol NamedSymbol && type.Is(KnownType.System_Func_T1_T2_T3_T4_TResult))
+            {
+                //HttpClientHandler.ServerCertificateCustomValidationCallback uses Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                //We're actually looking for Func<Any Sender, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                var parameters = NamedSymbol.DelegateInvokeMethod.Parameters;
+                return parameters.Length == 4   //And it should! T1, T2, T3, T4
+                    && parameters[0].Type.IsClassOrStruct() //We don't care about common (Object) nor specific (HttpRequestMessage) type of Sender
+                    && parameters[1].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Certificate2)
+                    && parameters[2].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Chain)
+                    && parameters[3].Type.Is(KnownType.System_Net_Security_SslPolicyErrors)
+                    && NamedSymbol.DelegateInvokeMethod.ReturnType.Is(KnownType.System_Boolean);
+            }
+            return false;
+        }
+
+        private static ImmutableArray<Location> ArgumentLocations(InspectionContext c, ExpressionSyntax expression)
+        {
+            var ret = ImmutableArray.CreateBuilder<Location>();
+            switch (expression)
+            {
+                case IdentifierNameSyntax identifier:
+                    var identSymbol = c.Context.SemanticModel.GetSymbolInfo(identifier).Symbol;
+                    if (identSymbol != null && identSymbol.DeclaringSyntaxReferences.Length == 1)
```

Comment:
_(subjective)_ you could skip the need of a `var` declaration by doing in the if
```
if (c.Context.SemanticModel.GetSymbolInfo(identifier).Symbol is ISymbol identSymbol &&
  identSymbol.DeclaringSyntaxReferences.Length == 1)
```

Reply:
Christope has the same note and it has been discussed [here](https://github.com/SonarSource/sonar-dotnet/pull/2785#discussion_r346245941)